eng/pipelines/templates/jobs/archetype-sdk-tests-jobs.yml

```diff
@@ -0,0 +1,127 @@
+parameters:
+- name: Platforms
+  type: object
+  default: []
+- name: PreSteps
+  type: object
+  default: []
+- name: PostSteps
+  type: object
+  default: []
+- name: EnvVars
+  type: object
+  default: {}
+- name: MaxParallel
+  type: number
+  default: 0
+- name: BuildInParallel
+  type: boolean
+  default: true
+- name: TimeoutInMinutes
+  type: number
+  default: 60
+- name: Location
+  type: string
+  default: ''
+- name: SubscriptionConfiguration
+  type: string
+  default: $(sub-config-azure-cloud-test-resources)
+- name: ServiceDirectory
+  type: string
+  default: not-specified
+
+jobs:
+  - ${{ each platform in parameters.Platforms }}:
+    - job:
+      condition: eq(eq(variables['Record'], 'true'), eq('${{ platform.TestMode }}', 'Record'))
```

Comment:
Why are you conditioning the entire job here?

Reply:
Because Recording is a separate job, we either run it for recording runs or skip for normal runs.

Reply:
Ok not entirely sure if this works but it might be a little easier to read this condition if you did `eq(variables['${{ platform.TestMode }}'], 'true')`. I'm thinking about having potential other conditional jobs in the future and we might be able to reuse that pattern.

Reply:
@weshaggard Due to the bug I've showed you yesterday, it looks like we can't use compile-time comparison.
The only thing I'd probably like to be generalized is the name of the variable, so that it can be used for other purposes. Something like:
```
or(and(not(variables['Category']), ${{ not(platform.Category) }}, eq('${{ platform.Category }}', variables['Category']))
```